nisystemlink/cross-service/GenericApi.py

```diff
@@ -0,0 +1,1040 @@
+        for column in dataresponse["columns"]:
+            column_names.append(column["name"])
+        return column_names
+
+    async def get_test_meta(self, record_id: str, keys: List[str] = None) -> Dict[str, str]:
+        """Gets some or all fo the metadata associated with a test.
+
+        :param record_id: A string representing the internal SystemLink id of a specific test
+        :param keys: a list describing the keys to return values for
+        """
+        resp = await self._tm_result_handle.get_result_v2(
+            result_id=record_id, _preload_content=False
+        )
+        response = await resp.read()
+        resp = json.loads(response)
+        name_map = {
+            "fileIds": "file_ids",
+            "hostName": "host_name",
+            "id": "id",
+            "keywords": "keywords",
+            "operator": "operator",
+            "partNumber": "part_number",
+            "programName": "program_name",
+            "properties": "properties",
+            "serialNumber": "serial_number",
+            "startedAt": "started_at",
+            "status": "status",
+            "statusTypeSummary": "status_type_summary",
+            "systemId": "system_id",
+            "totalTimeInSeconds": "total_time_in_seconds",
+            "updatedAt": "updated_at",
+            "workspace": "workspace",
+            "dataTableIds": "data_table_ids",
+        }
+        resp = {name_map[name]: val for name, val in resp.items()}
+        status_map = {"statusType": "status_type", "statusName": "status_name"}
+        resp["status"] = {status_map[name]: val for name, val in resp["status"].items()}
+        return_dict = {}
+        if keys:
+            for key in keys:
+                if key == "status":
+                    return_dict["status"] = resp["status"]["status_name"]
+                else:
+                    try:
+                        return_dict[key] = resp[key]
+                    except Exception:
+                        try:
+                            return_dict[key] = resp["properties"][key]
+                        except Exception:
+                            return_dict[key] = "Not Found"
+        else:
+            for key in resp:
+                #                 if key in ["started_at", "updated_at"]:
+                #                     temp = resp[key].isoformat()
+                #                     return_dict[key] = str(temp)
+                if key == "properties":
+                    for property in resp["properties"]:
+                        return_dict[property] = resp["properties"][property]
+                elif key == "status":
+                    return_dict["status"] = resp["status"]["status_name"]
+                elif key in ["status_type_summary", "id", "file_ids", "workspace"]:
+                    continue
+                else:
+                    return_dict[key] = resp[key]
+        return return_dict
+
+    async def get_channel_name_by_synonym(
```

Comment:
This method seems to be based on a custom `synonym` property. Since it's not a first-class part of the SystemLink API, behavior like this should live in a different package (in my opinion)